hibernate-core/src/main/java/org/hibernate/hql/internal/ast/tree/FromElement.java

```diff
@@ -678,23 +679,50 @@ public boolean isDereferencedBySubclassProperty() {
 
 	// ParameterContainer impl ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 	private List<ParameterSpecification> embeddedParameters;
+	private List<ParameterSpecification> embeddedFilterParameters;
```

Comment:
I think we can simply make it work with a single structure and we do the ordering in the getEmbeddedParameters method.